Core/src/org/sleuthkit/autopsy/casemodule/LocalDiskDSProcessor.java

```diff
@@ -221,6 +234,32 @@ public void reset() {
         ignoreFatOrphanFiles = false;
         setDataSourceOptionsCalled = false;
     }
+    
+    private static class AddLocalDiskCallbacks implements AddDataSourceCallbacks {
+        private final IngestStream ingestStream;
+        
+        AddLocalDiskCallbacks(IngestStream stream) {
+            ingestStream = stream;
+        }
+
+        @Override
+        public void onDataSourceAdded(long dataSourceObjectId) throws AddDataSourceCallbacksException {
+            try {
+                ingestStream.addDataSource(dataSourceObjectId);
+            } catch (IngestStreamClosedException ex) {
+                throw new AddDataSourceCallbacksException("Error adding files to ingest stream - ingest stream is closed", ex);
+            }
+        }
+        
+        @Override
+        public void onFilesAdded(List<Long> fileObjectIds) throws AddDataSourceCallbacksException {
+            try {
+                ingestStream.addFiles(fileObjectIds);
+            } catch (IngestStreamClosedException ex) {
+                throw new AddDataSourceCallbacksException("Error adding files to ingest stream - ingest stream is closed", ex);
+            }
+        }
+    }
```

Comment:
It looks like this class is the same as the one in ImageDSProcessor. We should probably have just one version of the class that is used by both DSPs.